imap_processing/ultra/l1a/ultra_l1a.py

```diff
@@ -28,7 +31,7 @@ def xarray_aux(decom_ultra_aux: dict):
     """
     # Converted time
     time_converted = []
-    for time in decom_ultra_aux["SHCOARSE"]:
+    for time in decom_ultra["SHCOARSE"]:
```

Comment:
We had discussed this in our meeting on monday, but Epoch must always be monotonically increasing.  So you can't have duplicate epoch numbers in your variable, and you'd unfortunately have to make a separate dimension to hold them.